_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,493 @@
+
+```
+KafkaClientsResource.deployKafkaClients(false, CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS).done();
+```
+
+##### Internal using client-examples image
+
+The second one is based on client-examples images, where everything about client implementation with also `Dockerfiles`
+can be found [here](https://github.com/strimzi/client-examples).
+This type of client is used as Job. For instance to send some messages provide following code:
+
+```
+KubernetesResource.deployNewJob(new JobBuilder()
+    .withNewMetadata()
+        .withNamespace(ResourceManager.kubeClient().getNamespace())
+        .withLabels(producerLabels)
+        .withName(producerName)
+    .endMetadata()
+    .withNewSpec()
+        .withNewTemplate()
+            .withNewMetadata()
+                .withLabels(producerLabels)
+            .endMetadata()
+            .withNewSpec()
+                .withRestartPolicy("OnFailure")
+                .withContainers()
+                    .addNewContainer()
+                    .withName(producerName)
+                        .withImage("strimzi/hello-world-producer:latest")
+                            .addNewEnv()
+                                .withName("BOOTSTRAP_SERVERS")
+                                .withValue(bootstrapServer)
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("TOPIC")
+                                .withValue(topicName)
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("DELAY_MS")
+                                .withValue("1000")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("LOG_LEVEL")
+                                .withValue("DEBUG")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("MESSAGE_COUNT")
+                                .withValue(String.valueOf(messageCount))
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("PRODUCER_ACKS")
+                                .withValue("all")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("ADDITIONAL_CONFIG")
+                                .withValue(additionalConfig)
+                            .endEnv()
+                    .endContainer()
+                .endSpec()
+            .endTemplate()
+        .endSpec()
+        .build());
+```
+
+##### External using Vert.x client
+
+Lastly, external clients are implemented by supporting Vertx.io Verticles. This type of client has the most limitations for usage. Most notable is dependency on external listeners, which is caused by`platform dependency`. 
```

Comment:
Can you explain what the _platform dependency_ is? Loadbalancers, Routes, NodePorts etc.?